### ISSUE TITLE
failed to deserialize EnumSet with json without type

### ISSUE THREAD

**Issue by sgandon** [jdereg/json-io#120]
When deserializing a json string created using JsonWriter.TYPE set to false, the string cannot be deserialized if it contains an EnumSet, the following code in *com.cedarsoftware.util.io.ObjectResolver* show that if item.getType() returns null (expected cause no type is written) then classForName() call will fail and throw an exception.
The type could be derived from the real object field type but here the Class c does not have the generic type the information, the Field instance has it but not the Class.

```
 private Object getEnumSet(Class c, JsonObject<String, Object> jsonObj)
    {
        Object[] items = jsonObj.getArray();
        if (items == null || items.length == 0)
        {
            return newInstance(c, jsonObj);
        }
        JsonObject item = (JsonObject) items[0];
        String type = item.getType();
        Class enumClass = MetaUtils.classForName(type, reader.getClassLoader());
        EnumSet enumSet = null;
        for (Object objectItem : items)
        {
            item = (JsonObject) objectItem;
            Enum enumItem = (Enum) getEnum(enumClass, item);
            if (enumSet == null)
            {   // Lazy init the EnumSet
                enumSet = EnumSet.of(enumItem);
            }
            else
            {
                enumSet.add(enumItem);
            }
        }
        return enumSet;
    }

```

**Comment by jdereg** [comment]
This method would need a variant that takes a field - and then that version used in the appropriate places.  Probably not a simple smash-and-grab job, though.

**Comment by wwang-talend** [comment]
here do a fix try for that : https://github.com/jdereg/json-io/pull/156 seems.

**Comment by jdereg** [comment]
fixed in release 4.14.1
